lib/core/src/Cardano/Wallet/Primitive/Types/Tx.hs

```diff
@@ -438,6 +439,17 @@ toTxHistory :: TransactionInfo -> (Tx, TxMeta)
 toTxHistory info =
     (fromTransactionInfo info, txInfoMeta info)
 
+-- | Information about when a transaction was submitted to the local node.
+-- This is used for scheduling resubmissions.
+data LocalTxSubmissionStatus tx = LocalTxSubmissionStatus
+    { txId :: !(Hash "Tx")
+    , submittedTx :: !tx
+    , firstSubmission :: !SlotNo
+    -- ^ Time of first successful submission to the local node.
+    , latestSubmission :: !SlotNo
+    -- ^ Time of most recent resubmission attempt.
```

Comment:
Would it be worth to add also the future schedule?

Reply:
The schedule can be calculated pretty easily from this type.